samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/grading/MediaData.hbm.xml

```diff
@@ -13,7 +13,7 @@
     <many-to-one name="itemGradingData"
       class="org.sakaiproject.tool.assessment.data.dao.grading.ItemGradingData"
       column="ITEMGRADINGID" index="SAM_MEDIA_ITEMGRADING_I" />
-    <property name="media">
+    <property name="dbMedia">
```

Comment:
Basically, we make the "media" property smarter (with the logic of where to read/write the bytes), but we still need a way to write to the database directly (really only for the DAO). The layout of the packages/classes mean that we have to make the DB property public to do this.

When trying to preserve the interface (getMedia giving the bytes, regardless of source) and maintaining the direct mapping, Hibernate was calling getMedia during persistence. This was doing the read from content and writing to the DB -- the whole situation we were trying to avoid.

So, this "internal only" distinction has to be made to be able to preserve the MediaData interface for outside code.